UKHomeOffice/sgmr-service
# SPCR-387 Select people for a voyage

### AC / Description of changes
- added new select people form for voyage
- added new people manifest page for voyage
- enabled selection of people and submit to voyage
- on return to 'select people' list, if people already on voyage then selection for them is disabled
- display people on voyage on manifest page


### To Test

Scenario One - no saved people
- delete all the saved people from your account
- create a voyage up to pleasure craft
- enter a pleasure craft and click save and continue
- > you should be taken to the people manifest page (note none of the buttons on this page currently work)

Scenario Two - saved people
- add at least three saved people
- create a voyage up to pleasure craft
- enter a pleasure craft and click save and continue
- > you should be taken to a list of your saved people to select people for the voyage
- select two people
- click save and continue
- > you should be taken to the people manifest page with those two people selected
- click back
- > you should be taken to the list of saved people page
- > the two people already on your voyage should be checked and disabled

Scenario Three - save and exit
- create a voyage up to pleasure craft
- enter a pleasure craft and click save and continue
- > you should be taken to a list of your saved people to select people for the voyage
- select at least one person
- click save and exit
- > you should be taken to the dashboard
- open that voyage
- > you should be shown the check your answers page
- click save and continue to return to the people list
- > you should be taken to the list of saved people page
- > the people already on your voyage should be checked and disabled

Note: This PR has the code from #523 merged up to it now. View that PR for it's test instructions


### Notes
- you cannot click to view the manifest page from the check your answers page yet
- two different PR's had changed App.jsx and the initial rebase only applied once set of changes, the second set have been re-added in this PR (Account changes from #511 
- if you only have one pleasure craft this flow will not work, this is fixed in #523 

---
* remember to merge to feature branches not master *
